x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/validation_step/validation_step_wrapper.tsx

```diff
@@ -29,22 +38,79 @@ export const ValidationStepWrapper: FC<CreateAnalyticsStepProps> = ({
     warning: 0,
     success: 0,
   });
-
+  const [checksInProgress, setChecksInProgress] = useState<boolean>(false);
+  const [validationMessages, setValidationMessages] = useState<CalloutMessage[]>([]);
+  const [errorMessage, setErrorMessage] = useState<CalloutMessage | undefined>();
+  const { form, jobConfig, isAdvancedEditorEnabled } = state;
+  const { jobType, trainingPercent, numTopFeatureImportanceValues, numTopClasses, includes } = form;
   const showValidationStep = step === ANALYTICS_STEPS.VALIDATION;
   const showDetails = step !== ANALYTICS_STEPS.VALIDATION && stepActivated === true;
+  const {
+    dataFrameAnalytics: { validateDataFrameAnalytics },
+  } = useMlApiContext();
 
   const dataTestSubj = `mlAnalyticsCreateJobWizardValidationStepWrapper${
     showValidationStep ? ' active' : ''
   }${showDetails ? ' summary' : ''}`;
 
+  const runValidationChecks = async () => {
+    try {
+      const analyticsJobConfig = (isAdvancedEditorEnabled
+        ? jobConfig
+        : getJobConfigFromFormState(form)) as DataFrameAnalyticsConfig;
+      const validationResults: ValidateAnalyticsJobResponse = await validateDataFrameAnalytics(
+        analyticsJobConfig
+      );
+
+      const valSummary = { warning: 0, success: 0 };
+      validationResults.forEach((message) => {
+        if (message?.status === VALIDATION_STATUS.WARNING) {
+          valSummary.warning++;
+        } else if (message?.status === VALIDATION_STATUS.SUCCESS) {
+          valSummary.success++;
+        }
+      });
+
+      setValidationMessages(validationResults);
+      setValidationSummary(valSummary);
+      setChecksInProgress(false);
```

Comment:
You set an error message in `catch` statement, but you don't reset it in case of successful fetch afterwards. 
```suggestion
      setChecksInProgress(false);
      setErrorMessage(undefined);
```

Reply:
Updated in 2d7ccc146f7e1b7e86daa70627c839f7e79e637c